src/harness/certs/generate_fake_certs.sh

```diff
@@ -150,6 +180,60 @@ openssl ca -cert unknown_ca.cert -keyfile private/unknown_ca.key -in unknown_dev
+    -reqexts cbsd_req -config ../../../cert/openssl.cnf \
+    -out client_expired.csr -keyout client_expired.key \
+    -subj "/C=US/ST=District of Columbia/L=Washington/O=Wireless Innovation Forum/OU=www.wirelessinnovation.org/CN=SAS CBSD Example"
+openssl ca -cert cbsd_ca.cert -keyfile private/cbsd_ca.key -in client_expired.csr \
+    -out client_expired.cert -outdir ./root \
+    -policy policy_anything -extensions cbsd_req_sign -config ../../../cert/openssl.cnf \
+    -batch -notext -create_serial -utf8 -startdate 20150214120000Z -enddate 20160214120000Z -md sha384
+
+#Certificate for test case WINNF.FT.S.SCS.15 - inapplicable fields certificate presented during registration
+echo "\n\nGenerate 'inapplicable certificate for WINNF.FT.S.SCS.15' certificate/key"
+openssl ca -cert cbsd_ca.cert -keyfile private/cbsd_ca.key -in client.csr \
+    -out client_inapplicable.cert -outdir ./root \
+    -policy policy_anything -extensions cbsd_req_inapplicable_sign -config ../../../cert/openssl.cnf \
+    -batch -notext -create_serial -utf8 -days 1185 -md sha384
+
+#Certificate for test case WINNF.FT.S.SCS.16 -
+openssl ca -revoke cbsd_ca.cert -keyfile private/root_ca.key -cert root_ca.cert \
+     -config ../../../cert/openssl.cnf
+
+echo "\n\n Generate CRL for root_ca"
+openssl ca -gencrl -keyfile private/root_ca.key -cert root_ca.cert \
+     -config ../../../cert/openssl.cnf  -crlhours 1\
+     -out root/crl/root_ca.crl
+
+echo "\n\n Generate CRL for cbsd_ca"
+openssl ca -gencrl -keyfile private/cbsd_ca.key -cert cbsd_ca.cert \
+     -config ../../../cert/openssl.cnf -crlhours 1 \
+     -out root/crl/cbsd_ca.crl
+
+# Certificate for WINNF.FT.S.SCS.17,WINNF.FT.S.SCS.18 and WINNF.FT.S.SCS.19
+echo "Created short lived certificate for WINNF.FT.S.SCS.17"
+current_time=`date -u  +%y%m%d%H%M%SZ`
+offset=5
+enddate_value=$(date -u -d "now + $offset minutes" '+%y%m%d%H%M%SZ')
+# Generate normal operation short_lived_client certificate/key.
+echo "\n\nGenerate 'short_lived_client' certificate/key"
+openssl req -new -newkey rsa:2048 -nodes \
+    -reqexts cbsd_req -config ../../../cert/openssl.cnf \
+    -out short_lived_client.csr -keyout short_lived_client.key \
+    -subj "/C=US/ST=District of Columbia/L=Washington/O=Wireless Innovation Forum/OU=www.wirelessinnovation.org/CN=SAS CBSD Example"
+openssl ca -cert cbsd_ca.cert -keyfile private/cbsd_ca.key -in short_lived_client.csr \
+    -out short_lived_client.cert -outdir ./root \
+    -policy policy_anything -extensions cbsd_req_sign -config ../../../cert/openssl.cnf \
+    -batch -notext -create_serial -utf8 -days 1185 -md sha384 -enddate $enddate_value
+
+# Generate trusted CA bundle.
+echo "\n\nGenerate 'ca' bundle"
+cat cbsd_ca.cert sas_ca.cert root_ca.cert cbsd-ecc_ca.cert sas-ecc_ca.cert root-ecc_ca.cert > ca.cert
+
+echo "Appended crl and create new trusted chain that contains revoked CA"
+cat ca.cert root/crl/cbsd_ca.crl root/crl/root_ca.crl  > WINNF_FT_S_SCS_16_ca.cert
```

Comment:
How is the CA revoked?

Reply:
Currently   the CBSD intermediate CA is revoked using the root CA and the revoked list is   generated. However, this change has been removed as the associated test case   has been removed.     This approach will be changed when a CRL server is implemented as part of   SCS16, 17, 18 and 19.